sda/protocol.go

```diff
@@ -59,6 +59,10 @@ func ProtocolRegisterName(name string, protocol NewProtocol) ProtocolID {
 		protocols = make(map[ProtocolID]NewProtocol)
 		protocolNames = make(map[ProtocolID]string)
 	}
+	if _, exists := protocolNames[u]; exists {
+		log.Warn("Protocol", u, "already exists - not overwriting")
```

Comment:
If we want that behavior, then please justify it (by a comment or something). It's not obvious why it should be like this. 
Moreover, there are similar questions regarding the `Service`, the `StatusReporter` etc... Everything that is registerable; For those, we'll either do keep a consistent behavior or adjust and justify every independant behavior.